velox/functions/prestosql/aggregates/MinMaxByAggregates.cpp

```diff
@@ -100,39 +108,14 @@ struct AccumulatorTypeTraits {};
 
 template <typename T>
 struct AccumulatorTypeTraits<T, std::enable_if_t<isNumericOrDate<T>(), void>> {
-  using AccumulatorType = T;
+  using AccumulatorType = std::optional<T>;
```

Comment:
I'm not sure I fully understand what you said.
We only compare and try to update accumulator if input comparison value is not null for now. and we don't care whether group is null.

Reply:
I'm thinking that we currently have a bool null flag for each group. Do we need any bool flag as part of std::optional or can be just use the null flag. My understanding is that null flag is TRUE only if we haven't seen any rows yet.

Reply:
I'm not sure how much overhead it will bring, but I would like keep current Aggregate framework, let each function to implement specific logic by itself.

Reply:
You are right, but I think current implementation would more make sense, at least for me.

Reply:
I see. The extra flag costs 8 bytes per group if T is int64_t, which doubles memory usage. 

sizeof(int64_t) is 8
sizeof(std::optional(int64_t)) is 16

For high cardinality aggregations this may be quite visible.

Reply:
> I'm thinking that we currently have a bool null flag for each group. Do we need another bool flag as part of std::optional or can be just use the null flag. My understanding is that null flag is TRUE only if we haven't seen any rows yet.

I understand what you mean now, newest modification has been pushed, thanks!